src/consensus/encode.rs

```diff
@@ -481,15 +484,22 @@ impl<T: Encodable> Encodable for Vec<T> {
 impl<T: Decodable> Decodable for Vec<T> {
     #[inline]
     fn consensus_decode<R: io::Read + ?Sized>(r: &mut R) -> Result<Self, Error> {
```

Comment:
Isn't this function doing the same thing as before, just now it won't compile on non-64 bit systems?

Reply:
I think it would be best to leave this function how it was and change Decodable for Box<[T]>  to use the same size checking as here before.

Reply:
It is a bit of a catch-22, as with
```rust
let len = usize::try_from(*VarInt::consensus_decode(r)?)
             .map_err(|_| Error::ParseFailed("VarInt overflows usize"))?;
```

`*VarInt::consensus_decode(r)?` can fail and can be tested but `usize::try_from(u64)` cannot be induced to fail and cannot be tested, thus contributing to failing the code coverage target, so I opted to restrict it to 64-bit systems.

However, I can gladly add the `usize::try_from(u64)` back in. What do you think?

Reply:
Ok, will change